### PR TITLE
Reworked grid layout and scaling + added persistence 

### PR DESCRIPTION
**Pull request by elsoazemelet**
Reworked grid layout (middle panel) in code and fixed scaling issue, with added persistence.

Grid Layout (middle panel)
- [x] Components are visually at their designated place (active changes bar, tracker bar, the rendered modules, etc.)
- [x] Logger messages and Tracker can not be shown at the same time
- [x] Engine Enabled/Disabled (on active changes bar) is working
- [x] Grids are correctly rendered in different shapes (try with a few modules)

Preferences

- [x] Module scaling works as intended
- [x] Module scaling is persistent now

### CODE REVIEW

**Comment by kkerti** [comment]
@narayb Possible bug here but may be unrelated: engine cannot be turned back to enabled, once it's disabled. I suspect the "Interacion Tracking" has different z-index, elements overlap.

**Comment by kkerti** [comment]
> @narayb Possible bug here but may be unrelated: engine cannot be turned back to enabled, once it's disabled. I suspect the "Interacion Tracking" has different z-index, elements overlap.

I see now in the checklist, that this pull request should fix what I mentioned. I did not check the pull request, only the release version, Sorry!

File: src/renderer/main/user-interface/ActiveChanges.svelte [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    engine,
+    runtime,
+    unsaved_changes,
+  } from "../../runtime/runtime.store";
+  import { writeBuffer } from "../../runtime/engine.store";
+  import { Analytics } from "../../runtime/analytics.js";
+  import { writable, derived, get } from "svelte/store";
+  import instructions from "../../serialport/instructions";
+
+  const isStoreEnabled = writable(false);
```
Comment (kkerti) [inline src/renderer/main/user-interface/ActiveChanges.svelte]: Ez miért `writable()`?

File: src/renderer/main/user-interface/ConnectModule.svelte
Comment (kkerti) [inline src/renderer/main/user-interface/ConnectModule.svelte:1]: Ennek találjunk jobb nevet, azt hiszem hogy ez modul csatlakozást kezelő componens, de csak egy dialog.

File: src/renderer/main/panels/configuration/components/Pages.svelte [hunk 2]
```diff
@@ -58,218 +32,36 @@
...
-  </div>
+    </button>
+  {/each}
 </page-controller>
 
 <style>
-  .dummy {
-    @apply bg-select;
-    @apply bg-opacity-50;
+  .grid-last-auto {
```
Comment (kkerti) [inline src/renderer/main/panels/configuration/components/Pages.svelte]: tailwindcss-ből nem lehetett ezt megoldani?

File: src/renderer/config-blocks/LedPhase.svelte [hunk 3]
```diff
@@ -170,6 +170,7 @@
           }}
           on:change={(e) => {
             let newValue = e.detail;
+            console.log("yay");
```
Comment (kkerti) [inline src/renderer/config-blocks/LedPhase.svelte]: Delete this

File: src/renderer/App.svelte [hunk 4]
```diff
@@ -206,6 +245,12 @@
 </main>
 
 <style global>
+  .grid-layout-container {
```
Comment (kkerti) [inline src/renderer/App.svelte]: Ez magic class vagy kell valahol? Tailwind?

**Review by kkerti — COMMENTED** [review]
@elsoazemelet beszéljünk a kódról két szót merge előtt és merge conflictot oldd fel.

**Comment by kkerti** [comment]
- If one module is plugged in, it comes in from bottom right. - appear in center
- If last module is plugged out, it goes out on bottom right. - disappear from center
- On module rotation, previous module UI is "cached", only upon interaction changes the module to correct view. - MAY BE FIRMWARE RELATED?
- Grids are correctly rendered... yes, but z-indexes are out of control: titlebar, active changes, page changes
image
- UNRELATED - On mac OS Venuta (M1) when grid editor is closed it cannot be reopened from dock.
image

File: src/renderer/App.svelte (hunk 4 above)
Comment (elsoazemelet) [reply src/renderer/App.svelte]: Bentmaradt kód.

File: src/renderer/main/user-interface/ActiveChanges.svelte (hunk 1 above)
Comment (elsoazemelet) [reply src/renderer/main/user-interface/ActiveChanges.svelte]: Fixed

File: src/renderer/main/user-interface/ConnectModule.svelte
Comment (elsoazemelet) [reply src/renderer/main/user-interface/ConnectModule.svelte:1]: Fixed

File: src/renderer/main/panels/configuration/components/Pages.svelte (hunk 2 above)
Comment (elsoazemelet) [reply src/renderer/main/panels/configuration/components/Pages.svelte]: Fixed

**Review by kkerti — APPROVED** [review]
Thank you @elsoazemelet ! Looks great.

**Pull request merged**